### PR TITLE
 Add helper functions to get inline css style for bg image in STARTER

### PR DESCRIPTION
**Pull request by andrew-147**

### CODE REVIEW

File: STARTER/STARTER.theme [hunk 1]
```diff
@@ -1,9 +1,57 @@
 <?php
```
Comment (T2L) [inline STARTER/STARTER.theme]: Move this to draft_theme.theme, as our theme will be a subtheme - so parent code will be available

File: STARTER/STARTER.theme [hunk 2]
```diff
@@ -1,9 +1,57 @@
...
 
 /**
- * Include all files from the includes directory
+ * @file
+ * STARTER Drupal 8 theme based on Draft theme.
```
Comment (T2L) [inline STARTER/STARTER.theme]: Fix comment as this will go in draft

File: STARTER/STARTER.theme [hunk 3]
```diff
@@ -1,9 +1,57 @@
...
-foreach (glob($includes_path) as $filename) {
-  require_once dirname(__FILE__) . '/includes/' . basename($filename);
+
+use Drupal\file\Entity\File;
+use Drupal\image\Entity\ImageStyle;
+
+/**
+ * Utility function to get the inline css style for a background image.
+ *
+ * @param mixed $entity
```
Comment (T2L) [inline STARTER/STARTER.theme]: Replace `mixed` with `Drupal\Core\Entity\ContentEntityInterface`

File: STARTER/STARTER.theme [hunk 4]
```diff
@@ -1,9 +1,57 @@
...
+ *   Entity containing the image field.
+ * @param string $field_name
+ *   Name of the image field to use.
+ * @param string $image_style
+ *   Name of the image style to use.
+ *
+ * @return string
+ *   Full css style attribute containing background image.
+ */
+function _get_entity_image_inline_style($entity, $field_name, $image_style) {
```
Comment (T2L) [inline STARTER/STARTER.theme]: Prefix with `draft_` and add $field_image_name (defaults to 'field_image')

File: STARTER/STARTER.theme [hunk 5]
```diff
@@ -1,9 +1,57 @@
...
+  $rendered_style = '';
```
Comment (T2L) [inline STARTER/STARTER.theme]: rename to `$output`

File: STARTER/STARTER.theme [hunk 6]
```diff
@@ -1,9 +1,57 @@
...
+
+  if (!$entity->$field_name->isEmpty()) {
```
Comment (T2L) [inline STARTER/STARTER.theme]: We might want to raise `InvalidArgumentException` when no field found. Let's think about it

File: STARTER/STARTER.theme [hunk 7]
```diff
@@ -1,9 +1,57 @@
...
+    $file_entity = $entity->get($field_name)->entity->get('field_image')->entity;
+    if ($url = _get_image_style_url($file_entity, $image_style)) {
+      $rendered_style = 'style="background-image: url(' . $url . ');"';
```
Comment (T2L) [inline STARTER/STARTER.theme]: Return plain url, constructing style attribute is template responsibility

File: STARTER/STARTER.theme [hunk 8]
```diff
@@ -1,9 +1,57 @@
...
+    }
+  }
+
+  return $rendered_style;
+}
+
+/**
+ * Get the url for a specific image style of an image.
+ *
+ * @param \Drupal\file\Entity\File $file
```
Comment (T2L) [inline STARTER/STARTER.theme]: Replace with `Drupal\file\FileInterface`

File: STARTER/STARTER.theme [hunk 9]
```diff
@@ -1,9 +1,57 @@
...
+ * @param string $image_style
+ *   Name of the image style to use.
+ *
+ * @return string
+ *   Url.
+ */
+function _get_image_style_url(File $file, $image_style) {
+  $uri = $file->getFileUri();
+  if ($image_style) {
+    $url = ImageStyle::load($image_style)->buildUrl($uri);
```
Comment (T2L) [inline STARTER/STARTER.theme]: No check for existing image style, raise `InvalidArgumentException`

File: STARTER/STARTER.theme [hunk 10]
```diff
@@ -1,9 +1,57 @@
...
+ *
+ * @param \Drupal\file\Entity\File $file
+ *   Image to be used.
+ * @param string $image_style
+ *   Name of the image style to use.
+ *
+ * @return string
+ *   Url.
+ */
+function _get_image_style_url(File $file, $image_style) {
```
Comment (T2L) [inline STARTER/STARTER.theme]: $image_style should be optional, i.e. `$image_style = NULL`

**Review by T2L — CHANGES_REQUESTED** [review]

File: STARTER/STARTER.theme (hunk 1 above)
Comment (andrew-147) [reply STARTER/STARTER.theme]: Done.

File: STARTER/STARTER.theme (hunk 2 above)
Comment (andrew-147) [reply STARTER/STARTER.theme]: Done.

File: STARTER/STARTER.theme (hunk 3 above)
Comment (andrew-147) [reply STARTER/STARTER.theme]: Done.

File: STARTER/STARTER.theme (hunk 4 above)
Comment (andrew-147) [reply STARTER/STARTER.theme]: Done.

File: STARTER/STARTER.theme (hunk 5 above)
Comment (andrew-147) [reply STARTER/STARTER.theme]: Done.

File: STARTER/STARTER.theme (hunk 7 above)
Comment (andrew-147) [reply STARTER/STARTER.theme]: Done.

File: STARTER/STARTER.theme (hunk 8 above)
Comment (andrew-147) [reply STARTER/STARTER.theme]: Done.

**Review by T2L — APPROVED** [review]

**Pull request merged**